lib/stats_reporter.rb

```diff
@@ -0,0 +1,117 @@
+# generates text for a report on storage root size and object status
+class StatsReporter
+
+  require 'text-table'
+  require 'faraday'
+  require 'open3'
+
+  REPOSITORY = 'sdr'.freeze
+  INGEST_WF = 'preservationIngestWF'.freeze
+  INGEST_WF_STEPS = [
+    'start-ingest',
+    'transfer-object',
+    'validate-bag',
+    'verify-apo',
+    'update-moab',
+    'update-catalog',
+    'complete-ingest'
+  ].freeze
+
+  def self.storage_report_text
+    head = %w[filesystem total used pct_used free pct_free]
+    Text::Table.new(head: head, rows: storage_report_lines).to_s
+  end
+
+  def workflow_report_text
+    head = %w[workflow waiting error recent archived]
+    Text::Table.new(head: head, rows: workflow_report_lines).to_s
+  end
+
+  def self.df_output
+    stdout_str, _status = Open3.capture2('df -h')
+    stdout_str
+  end
+
+  def self.storage_mount_lines
+    df_output.split("\n").select { |line| line.include?('services-disk') }
+  end
+
+  def self.parsed_storage_mount_lines
+    storage_mount_lines.map do |line|
+      storage_root = {}
+      split_line = line.split(' ')
+      storage_root[:total] = split_line[0]
+      storage_root[:used] = split_line[1]
+      storage_root[:remaining] = split_line[2]
+      storage_root[:percent_used] = split_line[3]
+      storage_root[:filesystem] = split_line[4]
+      storage_root[:percent_free] = "#{(100 - split_line[3].chop.to_i)}%"
+      storage_root
+    end
+  end
+
+  def self.storage_report_lines
+    parsed_storage_mount_lines.map do |h|
+      [h[:filesystem], h[:total], h[:used], h[:percent_used], h[:remaining], h[:percent_free]]
+    end
+  end
+
+  private_class_method :storage_mount_lines, :parsed_storage_mount_lines, :storage_report_lines
+
+  private
+
+  def ssl_options
+    {
+      client_cert: OpenSSL::X509::Certificate.new(File.read(Settings.ssl.cert_file)),
+      client_key:  OpenSSL::PKey::RSA.new(File.read(Settings.ssl.key_file), Settings.ssl.key_pass)
+    }
+  end
+
+  def conn
+    @conn ||= if ENV['ROBOT_ENVIRONMENT'] == 'test'
+                Faraday.new(url: Settings.workflow.url)
+              else
+                Faraday.new(url: Settings.workflow.url, ssl: ssl_options) do |f|
+                  f.adapter :net_http_persistent
+                end
+              end
+  end
+
+  def waiting_count
+    # in sdr-pc, this relied on register-sdr, which isn't a step in the WF
+    # so I'm guessing at the appropriate waiting parameter value
+    response = conn.get "workflow_queue?repository=#{REPOSITORY}&workflow=#{INGEST_WF}&waiting=start-ingest&count-only=true"
+    Nokogiri::XML(response.body).at_xpath('/objects/@count').value.to_i
```

Comment:
yea, i'm wondering if we can get this using workflow-services.  See https://github.com/sul-dlss/dor-workflow-service/blob/master/lib/dor/services/workflow_service.rb#L268.

As far as I can tell, this is fine.  It seems to correspond to https://github.com/sul-dlss/sdr-preservation-core/blob/master/bin/status_workflow.rb#L74  I would take out your comments.  :-)